examples/structured_data/classification_with_tfdf.py

```diff
@@ -0,0 +1,726 @@
+            data=tf.ones(
+                shape=(negative_feature_values.shape[0], 1), dtype=tf.dtypes.int32
+            ),
+            segment_ids=negative_feature_values,
+            num_segments=unique_feature_values.shape[0],
+        )
+
+        print("Target encoding: Storing target encoding statistics...")
+        self.positive_frequency_lookup = tf.constant(positive_frequency)
+        self.negative_frequency_lookup = tf.constant(negative_frequency)
+
+        self._is_adapted = True
+
+    def reset_state(self):
+        if self.built:
+            self.positive_frequency_lookup = None
+            self.negative_frequency_lookup = None
+
+    def call(self, inputs):
+        # data is expected to be a integer numpy array ot a Tensor shape [num_exmples, 1].
+        # This include the feature values for a given feature in the dataset.
+
+        # Raise an error of the target encoding statistics are not computed.
+        if (
+            self.positive_frequency_lookup == None
+            or self.negative_frequency_lookup == None
+        ):
+            raise ValueError(
+                f"You need to call the adapt method to compute target encoding statistics."
+            )
+
+        inputs = self._standardize_inputs(inputs)
+
+        # Fix and validate inputs shape.
+        if inputs.shape.rank == 1:
+            inputs = tf.expand_dims(inputs, axis=-1)
+        elif inputs.shape.rank > 2:
+            raise ValueError(
+                f"inputs is expected to be of rank 2, got {data.shape.rank}"
+            )
+        # Validate the dimensions of the second axis.
+        if inputs.shape[-1] != 1:
+            raise ValueError(
+                f"The second axis of inputs is expected to have 1 dimension, got {data.shape[-1]}"
+            )
+        # Lookup positive frequencies for the input feature values.
+        positive_fequency = tf.cast(
+            tf.gather_nd(self.positive_frequency_lookup, inputs),
+            dtype=tf.dtypes.float32,
+        )
+        # Lookup negative frequencies for the input feature values.
+        negative_fequency = tf.cast(
+            tf.gather_nd(self.negative_frequency_lookup, inputs),
+            dtype=tf.dtypes.float32,
+        )
+        # Compute positive probability for the input feature values.
+        positive_probability = positive_fequency / (
+            positive_fequency + negative_fequency
+        )
+        # Concatenate and return the looked-up statistics.
+        return layers.concatenate(
+            [positive_fequency, negative_fequency, positive_probability], axis=1
+        )
+
+    def compute_output_shape(self, input_shape):
+        # The output shape is expected to be [num_examplesm 3].
+        return tf.TensorShape([input_shape[0], 3])
+
+    def compute_output_signature(self, input_spec):
+        output_shape = self.compute_output_shape(input_spec.shape.as_list())
+        return tf.TensorSpec(shape=output_shape, dtype=tf.int64)
```

Comment:
There is some mixup with precision between the call and this method.

I would simply return a float32 for both the frequencies and ratios.